services/motion/builtin/builtin_test.go

```diff
@@ -599,7 +599,7 @@ func TestObstacleReplanning(t *testing.T) {
 		{
 			name: "ensure replan due to obstacle collision",
 			getPCfunc: func(ctx context.Context, cameraName string, extra map[string]interface{}) ([]*viz.Object, error) {
-				obstaclePosition := spatialmath.NewPoseFromPoint(r3.Vector{X: 1100, Y: 0, Z: 0})
+				obstaclePosition := spatialmath.NewPoseFromPoint(r3.Vector{X: 0, Y: 300, Z: 0})
```

Comment:
our goal is `geoPoint = (0,1e-5)`.
This means that in absolute terms the goal, as a pose, would be `(x,0,0)`.
We plan in relative terms so the goal, as a pose, would be `(0,y,0)` where `x == y`.

------------

On main, right now, if we were to add `fmt.Println("poseInPath: ", spatialmath.PoseToProtobuf(poseInPath))` to `CheckPlan`, we would see that our `poseInPath` value changes from `x:0` to `x:1111`, while `y` remains close to `0`.

------------
Further, we wish to have the obstacles of our worldstate which are fed into `CheckPlan` to be in their absolute position and not with respect to the base's frame which is what we currently have.
For this reason, I am curious as to why this change needs to take place.

------------

I am wondering if the reason as to why this change needed to be made is because we are prematurely calling `checkPlan`

Since we are using the fake PTG driver, there is a delay we hardcode when executing an input. 
Suppose we are executing inputs `[a,b,c]`, but we have not yet incremented out `wayPointIndex`. 
I am wondering if then, our current inputs would be `[a,b,c]`, while we believe the remainder of the plan still includes `[a,b,c]`. This could cause for us to incorrectly project our plan and might be the reason why you had to change the position of the obstacle in this test.

Reply:
Interesting.  Thanks for the detailed write-up.  I understand now why it was written this way but I am wondering if there is maybe another way to test this, since it seems like it is fragile to timing.  Instead of putting the obstacle at the goal, I am placing it directly in front of the robot `y=300`.  This should in the same way test if an obstacle is in the path correct?

Reply:
> but I am wondering if there is maybe another way to test this, since it seems like it is fragile to timing.

I agree that relying of timing is fragile.

> Instead of putting the obstacle at the goal, I am placing it directly in front of the robot y=300. 

What I am getting stuck on is: why is the obstacle at `y=300`, if it is was directly in front of the robot it should be at `x=300`, right?
If we place it in the Y axis, that is with respect to the base, when we want the position of the obstacle used for `CheckPlan` to be with respect to the word, i.e. in the X axis.

Reply:
Maybe I'm misunderstanding but I was taking the Y is forward and the fact that the obstacle is returned with respect to the base frame (or camera frame) that I should be putting it at Y is forward

Reply:
@nfranczak does my explanation make sense or is this version incorrect?

Reply:
Nick and I debugged this yesterday.  He agreed with my assessment that the test that currently exists is incorrect.  As such I would like to ticket this work out separately and add a `t.Skip()` in this PR since fixing this is outside the scope of this refactor.  Thoughts?

Reply:
test should be good now, that we have update `builtin_utils_test.go`